File: springboot-remoting-server/src/main/java/com/example/remoting/component/Runner.java
package com.example.remoting.component;

import com.example.remoting.component.Email;
import com.example.remoting.config.JmsServerConfig;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.jms.core.JmsTemplate;
import org.springframework.stereotype.Component;

@Slf4j
public class Runner implements CommandLineRunner {

  @Autowired
  JmsTemplate jmsTemplate;

  @Override
  public void run(String... args) throws Exception {
    log.info("Sending a email");
    jmsTemplate.convertAndSend(JmsServerConfig.DNS_QUERY_QUEUE, new Email("info@example.com", "Hello"));
  }
}

File: springboot-remoting-kotlin-server/src/main/kotlin/com/example/remoting/component/Runner.kt
package com.example.remoting.component

import com.example.remoting.config.JmsServerConfig
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.CommandLineRunner
import org.springframework.jms.core.JmsTemplate

class Runner : CommandLineRunner {
    companion object {
        val log: Logger = LoggerFactory.getLogger(Runner::class.java.name)
    }
    @Autowired
    private lateinit var jmsTemplate: JmsTemplate

    override fun run(vararg args: String?) {
        log.info("Send an email.")
        jmsTemplate.convertAndSend("dnsQueryServiceQueue1", Email("info@example.com", "Hello"))
    }
}